rust/xetblob/src/xet_repo.rs

```diff
@@ -291,22 +291,35 @@ impl XetRepo {
             .await
             .map_err(|_| anyhow::anyhow!("Branch does not exist"));
 
-        // Let's download all shards for now, delete this when shard hint lists work.
-        self.pull().await?;
-        sync_mdb_shards_from_git(
-            &self.config,
-            &self.config.merkledb_v2_cache,
-            GIT_NOTES_MERKLEDB_V2_REF_NAME,
-            true,
-        )
-        .await?;
-
         let mut transaction_config = self.config.clone();
         let shard_session_dir =
             TempDir::new_in(transaction_config.merkledb_v2_session, "mdb_session")?;
         transaction_config.merkledb_v2_session = shard_session_dir.path().to_owned();
 
         let translator = Arc::new(PointerFileTranslator::from_config(&transaction_config).await?);
+
+        // Download all shards for V1, by default none for V2.
+        let mut fetch_shards = false;
+
+        if let &PFTRouter::V1(_) = &self.translator.pft {
+            fetch_shards = true;
+        }
+
+        if let Some(v) = std::env::var_os("XET_FETCH_ALL_SHARDS") {
+            fetch_shards = v != "0";
+        }
+
+        if fetch_shards {
+            self.pull().await?;
+            sync_mdb_shards_from_git(
```

Comment:
This function only downloads shards. The above call to `self.pull()` pulls in `refs/notes/merkledb*` and merges into v1 MDB if the repo is v1. So we don't call this function but need to call `self.pull()` if `PFTRouter::V1(_) = &self.translator.pft`.

Reply:
I think the above logic calls both for MDB V1?

Reply:
Ah, I see what you mean.